src/blocks/slackBlocks.ts

```diff
@@ -13,9 +21,31 @@ export function markdown(text: string): MrkdwnElement {
   };
 }
 
-export function section(block: MrkdwnElement): KnownBlock {
+export function plaintext(text: string): PlainTextElement {
   return {
-    type: 'section',
-    text: block,
+    type: 'plain_text',
+    text,
   };
 }
+
+export function section(block?: MrkdwnElement, fields?: MrkdwnElement[]): KnownBlock {
+  if (block && fields) {
+    return {
+      type: 'section',
+      text: block,
+      fields: fields,
+    };
+  } else if (block) {
+    return {
+      type: 'section',
+      text: block,
+    };
+  } else if (fields) {
+    return {
+      type: 'section',
+      fields: fields,
+    };
+  } else {
+    throw new Error('Either block or fields must be provided.');
```

Comment:
Feels like this error should be thrown before undefined is passed in for both block and fields here.